{{cookiecutter.project_name}}/.github/workflows/tests.yml

```
@@ -30,43 +30,28 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
-      - name: Cache venv
+      - name: Cache tox
         uses: actions/cache@v3.2.6
-        id: cache_venv
+        id: cache_tox
         with:
           path: |
-            .venv
-          key: ${{ runner.os }}-venv-${{ matrix.python-version }}-${{ hashFiles('**/pyproject.toml') }}
+            .tox
+          key: ${{ runner.os }}-${{ matrix.python-version }}-tests
 
       - name: Set up Python
         uses: actions/setup-python@v4
-        id: setup_python
-        if: steps.cache_venv.outputs.cache-hit != 'true'
         with:
           python-version: ${{ matrix.python-version }}
 
       - name: Install dependencies
         shell: bash
         run: |
-          python3 -m venv .venv
-          if [ "${{ matrix.os }}" == "windows-latest" ]; then
-            source .venv/Scripts/activate
-          else
-            source .venv/bin/activate
-          fi
-          python3 -m pip install --upgrade pip
-          pip install -e .[tests]
+          pip install invoke tox
 
       - name: Run and write pytest
         shell: bash
         run: |
-          # activate venv dependent on the os
-          if [ "${{ matrix.os }}" == "windows-latest" ]; then
-            source .venv/Scripts/activate
-          else
-            source .venv/bin/activate
-          fi
-          pytest --durations=0 -n 2 -x --junitxml=pytest.xml --cov-report=term-missing --cov=src/ tests/
+          inv test --pytest-args="--durations=0" --pytest-args="--junitxml=pytest.xml --cov-report=term-missing --cov=src/ tests/"
```

two pytest args?

---

```suggestion
          inv test --pytest-args="--durations=0 --junitxml=pytest.xml --cov-report=term-missing --cov=src/"
```
It should autofind the tests folder - otherwise you might have the local test succeed and the CI fail.

---

> It should autofind the tests folder - otherwise you might have the local test succeed and the CI fail.

Agreed, it should not be at the github workflow level.

However, moving it to `inv test`. For moderately sized projects, specifying the tests dir can make test discovery ~1-2 s faster (e.g. `timeseriesflattener`).

---

I would say 1-2s is not worth it compared to the pain it would be to debug.

> Seems invoke has a bug where two pytest args are required for it to parse them as a list, rather than a string :S

I would probably just parse the string then, this seems like it could cause problems if the user does not know.